### PR TITLE
Bag of bugfixes

### PR DESCRIPTION
**Pull request by Chesteer89**

### CODE REVIEW

**Review by mateusz-cichosz — APPROVED** [review]

File: chat-components/database/src/main/java/com/pubnub/components/data/channel/DBChannel.kt [hunk 1]
```diff
@@ -15,13 +16,14 @@ data class DBChannel(
     @ColumnInfo(defaultValue = "default")
     override val type: String = "default",
     override val status: String? = null,
-    override val custom: ChannelCustomData? = null,
+    @field:TypeConverters(CustomContentConverter::class)
```
Comment (Chesteer89) [inline chat-components/database/src/main/java/com/pubnub/components/data/channel/DBChannel.kt:19]: @kazydek another fix for custom content. Now it's used everywhere

File: chat-components/database/src/main/java/com/pubnub/components/data/member/DBMember.kt [hunk 2]
```diff
@@ -17,15 +18,11 @@ data class DBMember(
     @ColumnInfo(defaultValue = "default")
     override val type: String = "default",
     override val status: String? = null,
-    override val custom: CustomData? = null,
+    @field:TypeConverters(CustomContentConverter::class)
```
Comment (Chesteer89) [inline chat-components/database/src/main/java/com/pubnub/components/data/member/DBMember.kt:21]: @kazydek another fix for custom content. Now it's used everywhere

File: chat-components/database/src/main/java/com/pubnub/components/repository/channel/DefaultChannelRepository.kt [hunk 3]
```diff
@@ -58,6 +58,8 @@ class DefaultChannelRepository(
 
         // Filtering
         if (filter != null) {
+            if (id != null)
```
Comment (Chesteer89) [inline chat-components/database/src/main/java/com/pubnub/components/repository/channel/DefaultChannelRepository.kt:61]: @kazydek not sure if we have it documented, if so please update this part :)

File: chat-components/database/src/main/java/com/pubnub/components/repository/member/DefaultMemberRepository.kt [hunk 4]
```diff
@@ -58,6 +58,8 @@ class DefaultMemberRepository(
 
         // Filtering
         if (filter != null) {
+            if (id != null)
```
Comment (Chesteer89) [inline chat-components/database/src/main/java/com/pubnub/components/repository/member/DefaultMemberRepository.kt:61]: @kazydek not sure if we have it documented, if so please update this part :)

File: chat-components/database/src/main/java/com/pubnub/components/repository/message/DefaultMessageRepository.kt [hunk 5]
```diff
@@ -95,6 +95,8 @@ class DefaultMessageRepository(
 
         // Filtering
         if (filter != null) {
+            if (id != null)
```
Comment (Chesteer89) [inline chat-components/database/src/main/java/com/pubnub/components/repository/message/DefaultMessageRepository.kt:98]: @kazydek not sure if we have it documented, if so please update this part :)

File: chat-components/src/main/java/com/pubnub/components/chat/network/mapper/NetworkChannelMapper.kt [hunk 6]
```diff
@@ -1,31 +1,16 @@
...
-            val gson = Gson()
-            val typeToken = object : TypeToken<ChannelCustomData>() {}.type
-            gson.fromJson(gson.toJson(custom), typeToken)
-        }
-
-
     override fun map(input: PNChannelMetadata): DBChannel {
-        val custom: ChannelCustomData? = getCustomData(input.custom)
-
+        val custom = input.custom.asMap()
```
Comment (Chesteer89) [inline chat-components/src/main/java/com/pubnub/components/chat/network/mapper/NetworkChannelMapper.kt:13]: @kazydek not sure if we have it documented, updated custom mapping

File: chat-components/src/main/java/com/pubnub/components/chat/provider/ChatProvider.kt [hunk 7]
```diff
@@ -117,6 +120,12 @@ fun ChatProvider(
         LocalUser provides pubNub.configuration.userId.value,
         LocalChannel provides channel,
 
+        // RTL support by locale
```
Comment (Chesteer89) [inline chat-components/src/main/java/com/pubnub/components/chat/provider/ChatProvider.kt:123]: @kazydek it's quite important. It forces application to use the system layout direction (left to right or right to left), depends on user locale

File: chat-components/src/main/java/com/pubnub/components/chat/ui/mapper/channel/DBChannelMapper.kt [hunk 8]
```diff
@@ -25,7 +26,7 @@ class DBChannelMapper : Mapper<DBChannelWithMembers, ChannelUi.Data> {
                 input.id,
                 input.name,
                 input.profileUrl,
-                input.custom?.description
+                input.custom.asMap()?.get("description") as? String?
```
Comment (Chesteer89) [inline chat-components/src/main/java/com/pubnub/components/chat/ui/mapper/channel/DBChannelMapper.kt:29]: @kazydek not sure if we have it documented, updated custom mapping

File: chat-components/src/main/java/com/pubnub/components/chat/ui/mapper/member/DBMemberMapper.kt [hunk 9]
```diff
@@ -2,9 +2,13 @@ package com.pubnub.components.chat.ui.mapper.member
...
+import com.pubnub.components.data.message.asMap
 import com.pubnub.framework.mapper.Mapper
 
 class DBMemberMapper : Mapper<DBMemberWithChannels, MemberUi.Data> {
     override fun map(input: DBMemberWithChannels): MemberUi.Data =
-        MemberUi.Data(input.id, input.name, input.profileUrl, input.member.custom?.description)
+        MemberUi.Data(input.id,
+            input.name,
+            input.profileUrl,
+            input.member.custom.asMap()?.get("description") as? String?)
```
Comment (Chesteer89) [inline chat-components/src/main/java/com/pubnub/components/chat/ui/mapper/member/DBMemberMapper.kt:13]: @kazydek not sure if we have it documented, updated custom mapping

File: chat-components/src/main/java/com/pubnub/components/chat/viewmodel/message/MessageViewModel.kt [hunk 10]
```diff
@@ -123,14 +124,18 @@ class MessageViewModel constructor(
      * @return Flow of Message UI Paging Data
      */
     fun getAll(
+        filter: Query? = null,
```
Comment (Chesteer89) [inline chat-components/src/main/java/com/pubnub/components/chat/viewmodel/message/MessageViewModel.kt:127]: @kazydek filter and sort method added, to allow customer use it on database level

**Pull request merged**